### PR TITLE
addressing feedback from @jepler

### PR DESCRIPTION
**Pull request by siddacious**
heya @jepler, this should address your feedback. I also left some comments on the google doc.

This is going to fail because I hate RST and am tired of futzing with it

### CODE REVIEW

**Comment by siddacious** [comment]
@jepler Were you able to test this?

File: adafruit_mcp2515/__init__.py [hunk 1]
```diff
@@ -222,24 +222,24 @@ def __init__(
         auto_restart: bool = False,
         debug: bool = False
     ):
-        """[summary]
+        """A common shared-bus protocol.
+        :param ~busio.SPI spi: The SPI bus used to communicate with the MCP2515
+        :param ~digitalio.DigitalInOut cs_pin:  SPI bus enable pin
+        :param int baudrate: The bit rate of the bus in Hz, using a 16Mhz crystal. All devices on the bus must agree on this value. Defaults to 250000.
+        :param bool loopback: Receive only packets sent from this device, and send only to this device. Requires that `silent` is also set to `False`, but only prevents transmission to other devices. Otherwise the send/receive behavior is normal.
```
Comment (jepler) [inline adafruit_mcp2515/__init__.py:229]: please update (requires that `silent` be `True`)

File: adafruit_mcp2515/__init__.py [hunk 2]
```diff
@@ -222,24 +222,24 @@ def __init__(
...
+
+        :param bool debug: If `True`, will enable printing debug information. Defaults to `False`.
         """
-        if loopback and silent:
-            raise AttributeError("only loopback or silent mode can bet set, not both")
+
+        if loopback and not silent:
+            raise AttributeError("Loopback mode requires silent to be set")
+        if auto_restart:
+            raise AttributeError("`auto-restart` is not supported by hardware")
```
Comment (jepler) [inline adafruit_mcp2515/__init__.py:241]: would it be useful to emulate this by restarting at any in_waiting(), receive() or send() call?  is the checking expensive (I could believe that)

File: adafruit_mcp2515/__init__.py [hunk 3]
```diff
@@ -324,33 +324,20 @@ def initialize(self):
...
          due to a full fifo or a bus error condition, RuntimeError is raised.
 
         Args:
             message (canio.Message): The message to send. Must be a valid `canio.Message`
         """
+
         # TODO: Timeout
         tx_buff = self._get_tx_buffer()  # info = addr.
+        if tx_buff is None:
+            raise RuntimeError("No transmit buffer available to send")
```
Comment (jepler) [inline adafruit_mcp2515/__init__.py:338]: OK this does implement what the core docs say.  It's not what core does.  sigh.  We can let some actual experience inform us as to what is preferable, and then implement it across the various chips, I don't want it to hold things back at this point.

File: adafruit_mcp2515/__init__.py [hunk 4]
```diff
@@ -841,6 +831,11 @@ def listen(self, matches=None, *, timeout: float = 10):
         """
         if matches is None:
             matches = []
+        elif self.silent and not self.loopback:
+            raise AttributeError(
+                "Hardware does not support setting `matches` in when\
+                `silent`==`True` and `loopback` == `False`"
```
Comment (jepler) [inline adafruit_mcp2515/__init__.py:837]: interesting restriction.

**Review by jepler — CHANGES_REQUESTED** [review]
I didn't re-test.

One doc change requested, and a general feeling of annoyance with myself about the semantics of send.  ugh I'm sorry about that.

File: adafruit_mcp2515/__init__.py (hunk 1 above)
Comment (siddacious) [reply adafruit_mcp2515/__init__.py]: fixed

File: adafruit_mcp2515/__init__.py (hunk 2 above)
Comment (siddacious) [reply adafruit_mcp2515/__init__.py:247]: I think it makes sense to defer implementation until its absence is raised as an issue

File: adafruit_mcp2515/__init__.py (hunk 3 above)
Comment (siddacious) [reply adafruit_mcp2515/__init__.py:344]: I agree that we should wait and see how things go.

File: adafruit_mcp2515/__init__.py (hunk 4 above)
Comment (siddacious) [reply adafruit_mcp2515/__init__.py:845]: I think they imagine that if you're in the monitoring mode that you want to "hear" everything?

**Review by jepler — APPROVED** [review]
I have totally lost the thread here.  Things look fine and if anything's not we can revisit later.

**Pull request merged**